src/components/Connections/CreateConnectionForm.tsx

```diff
@@ -51,670 +51,670 @@ interface SourceStream {
+                                  }
+                                }}
+                                renderValue={(selected: any) => selected.join(', ')}
+                              >
+                                {stream.primaryKeyConfig?.primaryKeyOptions?.length > 0 &&
+                                  stream.primaryKeyConfig.primaryKeyOptions.map(
+                                    (option: string, index: number) => (
+                                      <MenuItem key={option} value={option}>
+                                        <Checkbox
+                                          checked={stream.primaryKey.indexOf(option) > -1}
+                                          disabled={stream.primaryKeyConfig.sourceDefinedPrimaryKey}
+                                        />
+                                        {option}
+                                      </MenuItem>
+                                    )
+                                  )}
+                              </Select>
+                            </TableCell>
+                          </TableRow>
+                        );
+                      })}
+                  </TableBody>
+                </Table>
+              </>
+            )}
+          </Box>
+        </>
+      );
+    };
+
+    return (
+      <>
+        <CustomDialog
+          key={connectionId ? 'edit-custom' : 'new-custom'}
+          maxWidth={false}
+          data-testid="dialog"
+          title={connectionId ? 'Edit this connection' : 'Add a new connection'}
+          show={showForm}
+          handleClose={handleClose}
+          handleSubmit={handleSubmit(onSubmit)}
+          formContent={<FormContent />}
+          formActions={
+            <>
+              <Button variant="contained" type="submit" disabled={!someStreamSelected}>
+                Connect
+              </Button>
+              <Button color="secondary" variant="outlined" onClick={handleClose}>
+                Cancel
+              </Button>
             </>
-          )}
-        </Box>
+          }
+          loading={loading}
+        ></CustomDialog>
       </>
     );
-  };
-
-  return (
-    <>
-      <CustomDialog
-        key={connectionId ? 'edit-custom' : 'new-custom'}
-        maxWidth={false}
-        data-testid="dialog"
-        title={connectionId ? 'Edit this connection' : 'Add a new connection'}
-        show={showForm}
-        handleClose={handleClose}
-        handleSubmit={handleSubmit(onSubmit)}
-        formContent={<FormContent />}
-        formActions={
-          <>
-            <Button variant="contained" type="submit" disabled={!someStreamSelected}>
-              Connect
-            </Button>
-            <Button color="secondary" variant="outlined" onClick={handleClose}>
-              Cancel
-            </Button>
-          </>
-        }
-        loading={loading}
-      ></CustomDialog>
-    </>
-  );
-};
+  }
+);
 
 export default CreateConnectionForm;
```

Comment:
can add the `memo` here 
`export default memo(CreateConnectionForm)` just so that the diffs are more clear in this file.